docs/advanced.rst

```diff
@@ -133,10 +138,70 @@ To configure the participant to connect to another node through TCP, you must ad
+of scenarios, the involved network devices such as routers and firewalls should add the rules to allow the
+communication.
+
+For example, to allow incoming connections through our NAT, Fast-RTPS must be configured as a **TCP Server** listening
+to incoming TCP connections. To allow incoming connections through a WAN, the TCP descriptor associated must indicate
+its public IP through its field ``wan_addr``.
+
++--------------------------------------------------+
+| **C++**                                          |
++--------------------------------------------------+
+| .. literalinclude:: ../code/CodeTester.cpp       |
+|    :language: c++                                |
+|    :start-after: //CONF-TCP-TRANSPORT-SETTING    |
+|    :end-before: //!--                            |
++--------------------------------------------------+
+| **XML**                                          |
++--------------------------------------------------+
+| .. literalinclude:: ../code/XMLTester.xml        |
+|    :language: xml                                |
+|    :start-after: <!-->CONF-TCP-TRANSPORT-SETTING |
+|    :end-before: <!--><-->                        |
++--------------------------------------------------+
+
+In this case, configuring the router (which public IP is ``80.80.99.45``) is mandatory to allow the incoming traffic to
+reach the **TCP Server**. Typically a NAT routing with the ``listening_port`` ``5100`` to our machine is enough.
+Any existing firewall should be configured as well.
+
+In the client side, it is needed to specify the public IP of the **TCP Server** with its ``listening_port`` as
+``initial_peer``.
+
++---------------------------------------------------+
+| **C++**                                           |
++---------------------------------------------------+
+| .. literalinclude:: ../code/CodeTester.cpp        |
+|    :language: c++                                 |
+|    :start-after: //CONF-TCP2-TRANSPORT-SETTING    |
+|    :end-before: //!--                             |
++---------------------------------------------------+
+| **XML**                                           |
++---------------------------------------------------+
+| .. literalinclude:: ../code/XMLTester.xml         |
+|    :language: xml                                 |
+|    :start-after: <!-->CONF-TCP2-TRANSPORT-SETTING |
+|    :end-before: <!--><-->                         |
++---------------------------------------------------+
+
+The combination of the above configurations in both **TCP Server** and **TCP Client** allows a scenario similar to
+the represented by the following image.
+
+.. image:: TCP_WAN.png
+    :align: center
+
+Finally, the concepts **TCP Server** and **TCP Client** only refer to how the TCP communication is established and
+doesn't have any relationship with Fast-RTPS or RTPS concepts or **Publisher**, **Subscriber**, **Writer**, and
+**Reader**.
```

Comment:
```suggestion
**Reader**. Any of them can work as a **TCP Server** or a **TCP Client** because these entities are used
only to establish the TCP connection and the RTPS protocol works over it.
```